### ISSUE TITLE
IEMOCAP recipe - missing lengths in call to embedding module ?

### ISSUE THREAD

**Issue by wikong** [speechbrain/speechbrain#1028]
Hello Speechbrain team,

While trying to reproduce the results mentioned in https://github.com/speechbrain/speechbrain/tree/develop/recipes/IEMOCAP I figured out that the accuracy on the test set varies w.r.t the batch size in the test set dataloader, sometimes reducing the accuracy by up to 7% when batch_size = 1.

Further investigations in the code made me think that the 'lengths' parameter might be missing in the call to the embedding module (train.py line 37). 

I tried to change it to:
`embeddings = self.modules.embedding_model(feats, lengths=lens)`
and ran the train and test stages again.  Then the variations are less important but are still present : around 1% difference from one test to another (train stage done only once).

So I have 2 questions:
1. Do you think the modification hereabove is correct?
2. What could explain the remaining fluctuations with respect to the batch size of the test set?

Thank you in advance

**Comment by mravanelli** [comment]
I think the correct way is to pass the `lenghts`  parameter to the embedding model (this way zero padded elements are removed from the statistical pooling operation). @aheba, can we do it in your ongoing PR?

**Comment by curlneuron** [comment]
sorry，I didn't understand how to change this in the train_with_wav2vec2.yaml config file :

# The train logger writes training statistics to a file, as well as stdout.
train_logger: !new:speechbrain.utils.train_logger.FileTrainLogger
    save_file: !ref <train_log>

seed: 1993
__set_seed: !apply:torch.manual_seed [!ref <seed>]

Should I change this things after the "!"  ?
thx.
